Feature/issue 424

ret-Phoenix opened the pull request:
#424 СтрПолучитьСтроку работает с ошибками, исправил заменив цикл Split

ret-Phoenix committed:
Merge pull request #4 from EvilBeaver/develop

Взял обновления из основного репозитория

ret-Phoenix committed:
#424 СтрПолучитьСтроку работает с ошибками, исправил заменив цикл на Split

EvilBeaver commented:
А с какими ошибками он работал? Можем тест добавить?

ret-Phoenix committed:
Добавил провальный тест (при старом варианте)

EvilBeaver commented on tests/module.txt:
```diff
@@ -0,0 +1,105 @@
+// ������� ���� ������ ���������
+//
+// � �������� ������������� ����� �������� ������ �� ��������� ������� ������
+������� ��������������������(���������������, ������������� = ������������) �������
```
нужна кодировка UTF.

ret-Phoenix replied:
Хотел именно в оригинале показать, но сохранил в UTF8, та же ошибка, сейчас тогда еще для примера положу файл, который изначально рабочий и тест будет пройден

ret-Phoenix committed:
Пример в кодировке UTF8.
module-ok.txt - отрабатывает на обоих версиях движка
module.txt - на старом падает, на новом Ок.

ret-Phoenix commented:
Предлагаю ввести папку fixtures и класть туда файлы для тестовых наборов.

dmpas commented:
1. Если тестовый файл - это кусок модуля от 1С, то не стоит его включать - лицензия не одбряэ.
2. Тест можно сделать гораздо проще - достаточно любого текста, где первая строка пустая.

ret-Phoenix commented:
Только смоделировать ошибку смог на примере кода типовой 1С.

dmpas commented:
@ret-Phoenix 
```bsl
К = "
|1
|2
|3";

Сообщить("-" + СтрПолучитьСтроку(К, 1) + "-");
Сообщить("-" + СтрПолучитьСтроку(К, 2) + "-");
Сообщить("-" + СтрПолучитьСтроку(К, 3) + "-");
```

ret-Phoenix commented:
Да, подходит, на старом движке отрабатывает с ошибкой. на новом - Ок.

EvilBeaver commented:
@ret-Phoenix поправишь, заменив в примере код 1С на то, что предложил @dmpas?

ret-Phoenix committed:
Подправил тесты

ret-Phoenix committed:
Подправил тесты
